runtime/src/bank.rs

```diff
@@ -871,7 +869,15 @@ impl Bank {
         if parent.epoch() < new.epoch() {
             new.apply_feature_activations(false);
         }
+        let cloned = new
+            .stakes
+            .read()
+            .unwrap()
+            .clone_with_epoch(epoch, new.stake_program_v2_enabled());
+        *new.stakes.write().unwrap() = cloned;
 
+        let leader_schedule_epoch = epoch_schedule.get_leader_schedule_epoch(slot);
```

Comment:
moved here because we should update anything related stakes after we update the `Bank::stakes` conditionally on the runtime feature activation.